Feature/database/add guardians model

MahmoudSerag opened the pull request.

Commit:
Add guardian model
- Create guardian schema
- Add guardian validation messages
- Add relation between guardian and children

Commit:
Edit guardian schema
- Remove embedded document from childrenId property

Commit:
Edit guardian schema
- Add country code property, address property

IslamGoher commented on models/guardians.js:
```diff
@@ -0,0 +1,89 @@
+  fullName: {
+    type: String,
+    required: [true, 'Please add a name'],
+    minlength: 5,
+    maxlength: 50
```
please add validation message

IslamGoher commented on models/guardians.js:
```diff
@@ -0,0 +1,89 @@
+  password: {
+    type: String,
+    required: [true, 'Please add a password'],
+    minlength: 8,
+    maxlength: 64
```
please remove minlength and maxlength

IslamGoher commented on models/guardians.js:
```diff
@@ -0,0 +1,89 @@
+    minlength: 8,
+    maxlength: 64
+  },
+  picture: {
+    type: String,
+    required: false
+  },
```
please add url match

IslamGoher commented on models/guardians.js:
```diff
@@ -0,0 +1,89 @@
+        maxlength: 50,
+        required: [true, 'Please add a name on card']
+      },
+      phoneNumber: {
+        type: String,
+        required: [true, 'Please add a phone number']
+      },
```
please add match to phone number

IslamGoher reviewed (changes requested):
please fix those problems

Commit:
Edit guardians model
- Add validation message to fullName property
- Remove min and max length from password property
- Add url match to picture property
- Add match to phone number property

fix #7

Commit:
Merge branch 'develop' of https://github.com/Unknown-squad/toBeHero-backend into feature/database/add-guardians-model
Edit guardian model
